src/pants_bootstrap.rs

```diff
@@ -30,11 +30,27 @@ impl PantsBootstrap {
                 "-euo",
                 "pipefail",
                 "-c",
-                format!(
-                    r#"set >&2; source "{pants_bootstrap}" >"{capture}" 2>&1; set"#,
-                    pants_bootstrap = pants_bootstrap.display(),
-                    capture = capture.path().display(),
-                )
+                // N.B.: The `set` are used to print the current variables. The set -o posix are
+                // used to toggle restriction of `set` to printing only variable names and not more
+                // exotic items like functions. We ensure our modifications are unset before
+                // sourcing the .pants.bootstrap by using a sub-shell so that that script can use
+                // full bash-isms as has been the contract established in the original `./pants`
+                // script.
+                // N.B.: We take care to ensure the bash script here is a single line since the
+                // script contents are re-capitulated in the BASH_EXECUTION_STRING env var and a
+                // multiline env var will have its trailing lines fail to parse as env vars in most
+                // cases.
+                [
+                    r#"(set -o posix; IFS=$'\0'; set >&2); "#,
```

Comment:
I don't fully grok why the `set >&2` is needed as part of this. Does it still apply properly here when run in a sub-shell?

Reply:
The overall process is to diff a before and after set of shell variables. I use STDERR to capture before and STDOUT to capture after.